### ISSUE TITLE
Update Maven binary on Travis

### ISSUE THREAD

**Issue by marschall** [travis-ci/travis-ci#7719]
It look like the Maven version on Travis is 3.2.5 the current version is 3.5.0 https://maven.apache.org/download.cgi.

**Comment by chabala** [comment]
Maven 3.3.9 has been out [for more than a year](https://maven.apache.org/docs/history.html), 3.5.0 was recently released. Either would be better than 3.2.5. Would be nice to see this as a configurable option for the container based build system, like [the jdk options](https://docs.travis-ci.com/user/languages/java/#Testing-Against-Multiple-JDKs).

**Comment by marschall** [comment]
The JDKs are about as outdates as Maven, you get 1.8.0_31.

**Comment by rleigh-codelibre** [comment]
The new maven 3.5 is also required for building with JDK9, otherwise you get lots of odd errors, like

```
$ mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
Unrecognized VM option 'MaxPermSize=192m'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

or

```
% mvn
[ERROR] Error executing Maven.
[ERROR] java.lang.IllegalStateException: Unable to load cache item
[ERROR] Caused by: Unable to load cache item
[ERROR] Caused by: Could not initialize class com.google.inject.internal.cglib.core.$ReflectUtils
```

**Comment by TheSnoozer** [comment]
You can update the Maven-Install
see https://github.com/travis-ci/enterprise-installation/issues/14#issuecomment-273746437
worked fine for me

**Comment by chabala** [comment]
Looks like this has happened, my builds are running 3.5.0 now.

**Comment by marschall** [comment]
The Maven version is still outdated, current is 3.5.3 you provide 3.5.2.

**Comment by marschall** [comment]
I am happy to report that doing nothing did not fix the issue and Maven is still outdated.

**Comment by TheSnoozer** [comment]
Yes agree the version is outdated and potentially there will always be times where this is the case (simply due to the fact that updates needs to be tested, rolled out, etc).

In reality I would be curios what do you expect to change inside your build with the latest maven version?

If your product or your build really requires a certain version I would recommend to start using a custom installation routine (see https://github.com/travis-ci/enterprise-installation/issues/14#issuecomment-273746437 or the more complex version https://github.com/travis-ci/enterprise-installation/issues/14#issuecomment-329983790). The suggested workaround works still perfectly fine and does what I need (in my case I really want to test with different maven versions).

Just to note: by now maven-3.5.4 is released.

**Comment by agebhar1** [comment]
It's already updated to 3.5.3 via https://github.com/travis-ci/travis-cookbooks/pull/982 but unfortunately not available/rolled-out.

The latest update on build environment was at [2017/12/12](https://docs.travis-ci.com/user/build-environment-updates/2017-12-12/). Is there any ETA for a next update on the build environment?

**Comment by chabala** [comment]
I can confirm this is still an open issue. Running a container-based build configured for oraclejdk8 today picked up JDK 1.8.0_151-b12 and maven 3.5.2.

It looks like the common workaround is to force the installation of more current versions in before_install, but it's still a workaround for something that could be better fixed upstream of the user config.

**Comment by marschall** [comment]
I am currently having build failures due to the outdated Maven version:
https://travis-ci.org/marschall/memoryfilesystem/jobs/459111757

**Comment by cotsog** [comment]
Sorry if this is coming late but you can install Maven "manually" in your builds with a snippet like this in your .travis.yml file:

```
before_install:
  - wget http://apache.mirror.gtcomm.net/maven/maven-3/3.6.0/binaries/apache-maven-3.6.0-bin.tar.gz
  - tar xzvf apache-maven-3.6.0-bin.tar.gz
  - export PATH=`pwd`/apache-maven-3.6.0/bin:$PATH
  - mvn -v
```

This will install Maven 3.6.0. See sample build: https://travis-ci.org/cotsog/travis_ci_prod_regular/builds/481500766#L547.

Hope this helps!

**Comment by chabala** [comment]
You sure can @cotsog, but that effort is repeated for every build. It would be better to get Travis to support their infrastructure instead.

**Comment by mosabua** [comment]
You can also just use the Maven wrapper... https://github.com/takari/maven-wrapper .. just make sure the mvnw script is executable ..

**Comment by marschall** [comment]
I am happy to report that doing nothing did not fix the issue and Maven is still outdated.

**Comment by marschall** [comment]
Yup, still outdated but close now.

**Comment by marschall** [comment]
Still outdated.

**Comment by mosabua** [comment]
How about fixing the issues instead of just letting a bot try to close it.

**Comment by marschall** [comment]
Actually it seems to be fixed, the current version is Maven 3.6.3.

**Comment by mosabua** [comment]
Awesome. Then I guess this ticket can finally be closed    🎉
